app/src/main/java/com/github/se/travelpouch/model/activity/ActivityViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package com.github.se.travelpouch.model.activity
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.google.firebase.Firebase
+import com.google.firebase.firestore.firestore
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+class ActivityModelView(val activityRepositoryFirebase: ActivityRepositoryFirebase) : ViewModel() {
+
+    private val activities_ = MutableStateFlow<List<Activity>>(emptyList())
+    val activities: StateFlow<List<Activity>> = activities_.asStateFlow()
+
+    fun init() {
+        activityRepositoryFirebase.init { getActivities() }
+    }
+
+    fun getActivities() {
+        activityRepositoryFirebase.getActivity(onSuccess = { activities_.value = it }, onFailure = {})
```

Comment:
It would be helpful to add more context to the error log in the onFailure block:

```kotlin
onFailure = { Log.e("ActivityViewModel", "Failed to add travel", it) }